### ISSUE TITLE
[Android] Start crash

### ISSUE THREAD

**Issue by lipingruan** [wix/react-native-navigation#1275]
### Issue Description

APP Crashed on close system window animations

#### Screenshot
![screenshot_20170521-044409](https://cloud.githubusercontent.com/assets/16621637/26279243/e87f5efa-3de0-11e7-8ee2-690718ef178c.png)

#### Crash Log
<pre><code>
05-21 04:46:35.206 23131-23131/cn.sbsb.jsl E/WindowManager: android.view.WindowLeaked: Activity cn.sbsb.jsl.MainActivity has leaked window com.android.internal.policy.PhoneWindow$DecorView{c402646 V.E...... R....... 0,0-1080,1920} that was originally added here
                                                                at android.view.ViewRootImpl.<init>(ViewRootImpl.java:368)
                                                                at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:299)
                                                                at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:86)
                                                                at android.app.Dialog.show(Dialog.java:319)
                                                                at com.cboy.rn.splashscreen.SplashScreen$1.run(SplashScreen.java:34)
                                                                at android.app.Activity.runOnUiThread(Activity.java:5524)
                                                                at com.cboy.rn.splashscreen.SplashScreen.show(SplashScreen.java:24)
                                                                at com.cboy.rn.splashscreen.SplashScreen.show(SplashScreen.java:44)
                                                                at cn.sbsb.jsl.MainActivity.onCreate(MainActivity.java:26)
                                                                at android.app.Activity.performCreate(Activity.java:6251)
                                                                at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108)
                                                                at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2403)
                                                                at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2520)
                                                                at android.app.ActivityThread.-wrap11(ActivityThread.java)
                                                                at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1363)
                                                                at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                at android.os.Looper.loop(Looper.java:148)
                                                                at android.app.ActivityThread.main(ActivityThread.java:5466)
                                                                at java.lang.reflect.Method.invoke(Native Method)
                                                                at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
                                                                at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:102)
05-21 04:46:35.484 23131-23201/cn.sbsb.jsl W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.ReactVirtualTextShadowNode
05-21 04:46:35.567 23131-23131/cn.sbsb.jsl D/AndroidRuntime: Shutting down VM
                                                             
                                                             
                                                             --------- beginning of crash
05-21 04:46:35.568 23131-23131/cn.sbsb.jsl E/AndroidRuntime: FATAL EXCEPTION: main
                                                             Process: cn.sbsb.jsl, PID: 23131
                                                             java.lang.IllegalArgumentException: View=com.android.internal.policy.PhoneWindow$DecorView{c402646 V.E...... R....... 0,0-1080,1920} not attached to window manager
                                                                 at android.view.WindowManagerGlobal.findViewLocked(WindowManagerGlobal.java:424)
                                                                 at android.view.WindowManagerGlobal.removeView(WindowManagerGlobal.java:350)
                                                                 at android.view.WindowManagerImpl.removeViewImmediate(WindowManagerImpl.java:118)
                                                                 at android.app.Dialog.dismissDialog(Dialog.java:362)
                                                                 at android.app.Dialog.dismiss(Dialog.java:345)
                                                                 at com.cboy.rn.splashscreen.SplashScreen$2.run(SplashScreen.java:58)
                                                                 at android.os.Handler.handleCallback(Handler.java:739)
                                                                 at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                 at android.os.Looper.loop(Looper.java:148)
                                                                 at android.app.ActivityThread.main(ActivityThread.java:5466)
                                                                 at java.lang.reflect.Method.invoke(Native Method)
                                                                 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
                                                                 at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:102)

</code></pre>

---
### Environment
* React Native Navigation version: 2.0.0-experimental.294
* React Native version: 0.43.2
* Platform(s) (iOS, Android, or both?): Android
* Device info (Simulator/Device? OS version? Debug/Release?): Device, Android 6.0.1, Debug & Release

**Comment by hzyIO** [comment]
@lipingruan What did you do to fix this? I'm getting this same issue and I think its related to RNN.

**Comment by lipingruan** [comment]
@hzyIO no resolved : )

**Comment by lipingruan** [comment]
my android system [low battery mode] will auto close animations.

**Comment by rclaes1995** [comment]
I'm having this issue too, and I feel it may be a problem with the Splash Screen library rather than the navigation one. See this issue:
https://github.com/crazycodeboy/react-native-splash-screen/issues/69

**Comment by brunolemos** [comment]
I think this issue is some conflict between react-navigation and react-native-splash-screen.
I have this too. (https://github.com/crazycodeboy/react-native-splash-screen/issues/32#issuecomment-362114853)

Does anyone have a workaround?

**Comment by brunolemos** [comment]
Fix for WindowLeaked was mentioned at https://github.com/crazycodeboy/react-native-splash-screen/issues/32:

Add this method:

```java
    @Override
    protected void onPause() {
        SplashScreen.hide(this);
        super.onPause();
    }
```

**Comment by pqkluan** [comment]
Why don't you guys use the [supported Splash Screen](http://wix.github.io/react-native-navigation/#/android-specific-use-cases?id=splash-screen) instead of another lib?

**Comment by brunolemos** [comment]
Because react-native-splash-screen provides more control on when you want to hide it and show the app. e.g. only after redux rehydrate etc

**Comment by pqkluan** [comment]
You could use the supported splash screen without having to draw it in Java.
```java
public class MainActivity extends SplashActivity {
    @Override
    public int getSplashLayout() {
        return R.layout.splash;
    }
}
```

**Comment by brunolemos** [comment]
> You could use the supported splash screen without having to draw it in Java
return R.layout.splash;

Thanks, great to know 👍 
It would be nice to have this in the docs if it isn't already.

**Comment by brunolemos** [comment]
> The supported splash screen won't go away unless you call startTabBasedApp or startSingleScreenApp

For both Android and iOS I suppose?
I will remove react-native-splash-screen from the project then 👍
